### ISSUE TITLE
yabai -m display west|south|north|east quirk/bug with different sized displays

### ISSUE THREAD

**Issue by nerdo** [koekeishiya/yabai#1724]
I've noticed that if I use the `yabai -m display west` command from a display with smaller dimensions and it moves to a display with larger dimensions, the display seems to focus but it doesn't focus on a window in that display.

I thought maybe it had to do with spaces but it seems to happen somewhat consistently, and the only thing I can think of is maybe it has something to do with the sizes of the displays being different.

The only workaround I've found isn't really viable. If I switch displays with that command and then switch to another space and back, it seems to focus on the window that should have been focused in the first place.

It always seems to work in one direction though (e.g. east) but not the opposite direction (e.g. west) - this could still be a spaces issue?

**Comment by koekeishiya** [comment]
Is this still an issue?

**Comment by niba** [comment]
@koekeishiya  yes, i have 3 monitor setup and "west" is buggy, it jumps from center display to the right one (not "west"). It works okay with "east" movement

I have [laptop display] [27 inch monitor] [24 inch monitor] so it can be related to the size

**Comment by koekeishiya** [comment]
Probably caused by negative coordinates. Output of: yabai -m query --displays index,frame

**Comment by niba** [comment]
Here is my output:

```
[{
        "index":1,
        "frame":{
                "x":0.0000,
                "y":0.0000,
                "w":2560.0000,
                "h":1440.0000
        }
},{
        "index":2,
        "frame":{
                "x":-1728.0000,
                "y":0.0000,
                "w":1728.0000,
                "h":1117.0000
        }
},{
        "index":3,
        "frame":{
                "x":2560.0000,
                "y":0.0000,
                "w":1920.0000,
                "h":1080.0000
        }
}]
```

My monitor layout by index [2] [1] [3] so you are right and my laptop display has negative coordinates

**Comment by koekeishiya** [comment]
Hm so I just tested this part of the code with your display configuration and it returns the correct result, so I'm not sure what exactly the issue is here.

**Comment by koekeishiya** [comment]
Likely a variant of #1569 , #1718 , #1736
